TideApp/TideAppWidget/Helpers/TidesWidgetLocationManager.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  TidesWidgetLocationManager.swift
+//  TideApp
+//
+//  Created by Marco Guerrieri on 14/05/2021.
+//
+
+import Foundation
+import CoreLocation
+
+final class TidesWidgetLocationManager: NSObject, CLLocationManagerDelegate {
+  private var completion: ((CLLocation?)->())?
+  private let locationManager: CLLocationManager
```

Comment:
I dont really know much about how widgets work so this might not be possible, but could we use the `UserLocator` class here?

Reply:
Yeah, changed to use that! Tbh I tried before but I wasn't able to use it, but I had just to understand that for some reason I have to reject the first location received (I just filter for location with lat and long == 0 atm, this because also that kind of coordinates generate an error in the api call)